lib/Dialect/OM/Evaluator/Evaluator.cpp

```diff
@@ -591,6 +603,9 @@ circt::om::Evaluator::evaluateListConcat(ListConcatOp op,
       return result;
     if (!result.value()->isFullyEvaluated())
       return list;
+    // Return an unknown value if any unknown value is observed.
+    if (isa<evaluator::UnknownValue>(*result.value()))
```

Comment:
nit: code here seems to prefer `.get()` over `operator*`, for consistency please consider following the style/details of the surrounding code (I imagine this was mostly copied+edited, I get it).

Reply:
Yeah, fair. I was seeing a lot of `result->get()` when I subjectively favored the `*result`. I'll do a sweep of this and switch to `->get()` to make it consistent. Good thought.

Reply:
Fixed in: 748d18bef